exporting/exporting_engine.c

```diff
@@ -4,6 +4,55 @@
 
 static struct engine *engine = NULL;
 
+/**
+ * Exporting Clean Engine
+ *
+ * Clean all variables allocated inside engine structure
+ *
+ * @param en a pointer to the strcuture that will be cleaned.
+ */
+static void exporting_clean_engine()
+{
+    if (!engine)
+        return;
+
+    int disable_awk_sdk = 0;
+#if ENABLE_PROMETHEUS_REMOTE_WRITE
+    int disable_remote_write = 0;
```

Comment:
Why do you need `disable_awk_sdk` and `disable_remote_write `?
You can check `engine->aws_sdk_initialized` and add `protocol_buffers_initialized` to `struct engine` and set it to 1 after `init_write_request()` is called. You don't need to check them for every instance. You even don't need to reset them as engine cleaning is the last operation before the main thread exits.

Reply:
I brought the adjusts you requested, I do not have all the code and logic related to AWS and Prometheus in my mind yet, thank you for all tips related to these databases.